File: two-sums.py
def twoSum(num: list, target: int) -> list:
    a_pointer = 0
    b_pointer = len(num) - 1
    while a_pointer < b_pointer:
        sum = num[a_pointer] + num[b_pointer]
        if sum == target:
            # print (num[a_pointer], num[b_pointer])
            return num[a_pointer], num[b_pointer]
        elif sum > target:
            b_pointer = b_pointer - 1
        elif sum < target:
            a_pointer = a_pointer + 1


twoSum([2, 2, 3, 7, 11, 15], 9)

File: two-sums.ts
const twoSum = (num: Array<number>, target: number) => {
    let nums = num.slice()
    nums.sort(function (a, b) { return a - b });
    let a_pointer = 0;
    let b_pointer = nums.length - 1;
    while (a_pointer < b_pointer) {
        let sum = nums[a_pointer] + nums[b_pointer];
        if (sum === target) {
            return [num.indexOf(nums[a_pointer]), num.indexOf(nums[b_pointer])];
        } else if (sum > target) {
            b_pointer--;
        } else {
            a_pointer++;
        }
    }
}

console.log(twoSum([3, 3], 6))